esi_leap/resource_objects/base.py

```diff
@@ -41,7 +41,7 @@ def get_node_config(self):
         """Returns resource's associated config, if any"""
 
     @abc.abstractmethod
-    def get_resource_class(self):
+    def get_resource_class(self, resource_list):
```

Comment:
Was trying to figure out if this would enforce including resource_list as an argument in derived classes but it looks like it doesn't? Not sure if this is even all that important but I'll include it as a note https://stackoverflow.com/questions/22581974/abstract-methods-with-specific-arguments-in-python